libflux/go/libflux/analyze_test.go

```diff
@@ -56,3 +58,170 @@ func TestAnalyze(t *testing.T) {
+			{
+				Name:     "",
+				Metadata: "parser-type=rust",
+				Package: &ast.PackageClause{
+					Name: &ast.Identifier{Name: "foo"},
+				},
+				Body: []ast.Statement{
+					&ast.VariableAssignment{
+						ID:   &ast.Identifier{Name: "b"},
+						Init: &ast.IntegerLiteral{Value: 1},
+					},
+				},
+			},
+			{
+				Name:     "",
+				Metadata: "parser-type=rust",
+				Package: &ast.PackageClause{
+					Name: &ast.Identifier{Name: "foo"},
+				},
+				Body: []ast.Statement{
+					&ast.VariableAssignment{
+						ID:   &ast.Identifier{Name: "a"},
+						Init: &ast.IntegerLiteral{Value: 1},
+					},
+				},
+			},
+		},
+	}
+
+	if !cmp.Equal(got, want, asttest.IgnoreBaseNodeOptions...) {
+		t.Errorf(
+			"MergePackages unexpected packages -want/+got:\n %v",
+			cmp.Diff(want, got, asttest.IgnoreBaseNodeOptions...),
+		)
+	}
+}
+
+func TestNoInFiles(t *testing.T) {
+	pkg1 := libflux.Parse(`package foo`)
+	pkg2 := libflux.Parse(`
+package foo
+
+b = 1`)
+
+	err := libflux.MergePackages(pkg1, pkg2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	gotFB, err := pkg2.MarshalFB()
+	got := ast.DeserializeFromFlatBuffer(gotFB)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := &ast.Package{
+		Package: "foo",
+		Files: []*ast.File{
+			{
+				Name:     "",
+				Metadata: "parser-type=rust",
+				Package: &ast.PackageClause{
+					Name: &ast.Identifier{Name: "foo"},
+				},
+				Body: []ast.Statement{
+					&ast.VariableAssignment{
+						ID:   &ast.Identifier{Name: "b"},
+						Init: &ast.IntegerLiteral{Value: 1},
+					},
+				},
+			},
+			{
+				Name:     "",
+				Metadata: "parser-type=rust",
+				Package: &ast.PackageClause{
+					Name: &ast.Identifier{Name: "foo"},
+				},
+			},
+		},
+	}
+
+	if !cmp.Equal(got, want, asttest.IgnoreBaseNodeOptions...) {
+		t.Errorf(
+			"MergePackages unexpected packages -want/+got:\n %v",
+			cmp.Diff(want, got, asttest.IgnoreBaseNodeOptions...),
+		)
+	}
+}
+
+func TestNoOutFiles(t *testing.T) {
+	pkg1 := libflux.Parse(`
+package foo
+
+b = 1`)
+	pkg2 := libflux.Parse(`package foo`)
+
+	err := libflux.MergePackages(pkg1, pkg2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	gotFB, err := pkg2.MarshalFB()
+	got := ast.DeserializeFromFlatBuffer(gotFB)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := &ast.Package{
+		Package: "foo",
+		Files: []*ast.File{
+			{
+				Name:     "",
+				Metadata: "parser-type=rust",
+				Package: &ast.PackageClause{
+					Name: &ast.Identifier{Name: "foo"},
+				},
+			},
+			{
+				Name:     "",
+				Metadata: "parser-type=rust",
+				Package: &ast.PackageClause{
+					Name: &ast.Identifier{Name: "foo"},
+				},
+				Body: []ast.Statement{
+					&ast.VariableAssignment{
+						ID:   &ast.Identifier{Name: "b"},
+						Init: &ast.IntegerLiteral{Value: 1},
+					},
+				},
+			},
+		},
+	}
+
+	if !cmp.Equal(got, want, asttest.IgnoreBaseNodeOptions...) {
+		t.Errorf(
+			"MergePackages unexpected packages -want/+got:\n %v",
+			cmp.Diff(want, got, asttest.IgnoreBaseNodeOptions...),
+		)
+	}
+}
```

Comment:
It would be good to have some tests for the errors that can happen, like when package names do not match, etc.

Reply:
I have some test like that on the Rust side. I can add some similar tests here if that would be helpful.

Reply:
I think it would be good to test the error path.  There is some code on the Go side that will not get exercised otherwise.

Reply:
I made some changes to the test file. Let me know if you know of any other cases that I should also cover.